### ISSUE TITLE
torch.compile has different numerics for var_mean

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#145401]
### 🐛 Describe the bug

```
import torch
from torch._dynamo.utils import same

def foo(x):
    return torch.ops.aten.var_mean.correction(x, [1], correction = 0, keepdim = True)

inp = torch.rand([112958, 384], device="cuda", dtype=torch.float16)
print(same(foo(inp), torch.compile(foo)(inp)))
```

> [ERROR]:Accuracy failed: allclose not within tol=0.0001

Maybe this is a numerics sensitive op, but it throws up bisecting and is a general pain. 

cc @ezyang 

### Versions

master

**Comment by eellison** [comment]
Potentially add config to disable welford

**Comment by karthickai** [comment]
**🤖 PT2 Repro Agent**

I tried to reproduce this issue on PyTorch nightly (`2.13.0.dev20260516+cu130`) on NVIDIA H100 and confirmed it reproduces (3/3 runs failed).

All 3 runs failed deterministically with the same numerical mismatch reported in the issue: `Accuracy failed: allclose not within tol=0.0001` from `torch._dynamo.utils.same`. Eager vs compiled max diff on float16 input of shape [112958, 384]: var=6.10e-05 (passes atol=1e-4), mean=4.88e-04 (fails atol=1e-4). The mean discrepancy exceeds the default tolerance and drives the failure. The `emulate_precision_casts=True` mitigation was tested and did NOT fix the bug (same max diffs).

The compiler bisector identified the offending component:
- **Backend**: `inductor`
- **Subsystem**: `lowerings`
- **Detail**: `var_mean`

<details>
<summary>Per-subsystem coverage (9 tested)</summary>

| Subsystem | Bug fixed when disabled? |
|---|---|
| `pre_grad_passes` | ❌ no |
| `joint_graph_passes` | ❌ no |
| `post_grad_passes` | ❌ no |
| `inductor_fallback_random` | ❌ no |
| `inductor_emulate_precision_casts` | ❌ no |
| `inductor_layout_optimization` | ❌ no |
| `inductor_comprehensive_padding` | ❌ no |
| `cudagraphs` | ❌ no |
| `lowerings` | ✅ yes |

Replay: `TORCH_BISECT_BACKEND=inductor TORCH_BISECT_SUBSYSTEM=lowerings TORCH_BISECT_MAX=1 python repro.py`

_Tool: `torch._inductor.compiler_bisector`_
</details>

<details>
<summary>Error output</summary>

```
var max diff: 6.103515625e-05
mean max diff: 0.00048828125
  atol=0.0001: var=PASS, mean=FAIL
  atol=1e-05: var=FAIL, mean=FAIL
  atol=1e-06: var=FAIL, mean=FAIL
  atol=1e-07: var=FAIL, mean=FAIL
  atol=1e-08: var=FAIL, mean=FAIL

E0518 01:34:03.342000 1723726 site-packages/torch/_dynamo/utils.py:3619] Accuracy failed: allclose not within tol=0.0001

ERROR conda.cli.main_run:execute(49): `conda run python repro.py` failed. (See above for error)
```
</details>

<details>
<summary>Repro script I used</summary>

```python
import os
import torch
from torch._dynamo.utils import same


def foo(x):
    return torch.ops.aten.var_mean.correction(x, [1], correction=0, keepdim=True)


def test_fn() -> bool:
    """Return True if NO bug (eager == compiled). False if bug present."""
    torch._dynamo.reset()
    backend = os.environ.get("TORCH_COMPILE_BACKEND", "inductor")
    torch.manual_seed(0)
    inp = torch.rand([112958, 384], device="cuda", dtype=torch.float16)
    eager_out = foo(inp)
    compiled_out = torch.compile(foo, backend=backend)(inp)
    return bool(same(eager_out, compiled_out))


if __name__ == "__main__":
    device = "cuda" if torch.cuda.is_available() else "cpu"
    torch.manual_seed(0)
    inp = torch.rand([112958, 384], device=device, dtype=torch.float16)

    eager_var, eager_mean = foo(inp)
    compiled_var, compiled_mean = torch.compile(foo)(inp)

    var_diff = (eager_var.float() - compiled_var.float()).abs().max().item()
    mean_diff = (eager_mean.float() - compiled_mean.float()).abs().max().item()
    print(f"var max diff: {var_diff}")
    print(f"mean max diff: {mean_diff}")
    for atol in (1e-4, 1e-5, 1e-6, 1e-7, 1e-8):
        ok_var = torch.allclose(eager_var, compiled_var, atol=atol)
        ok_mean = torch.allclose(eager_mean, compiled_mean, atol=atol)
        print(f"  atol={atol}: var={'PASS' if ok_var else 'FAIL'}, mean={'PASS' if ok_mean else 'FAIL'}")

    try:
        ok = test_fn()
    except Exception:
        import traceback
        traceback.print_exc()
        os._exit(2)
    os._exit(0 if ok else 1)

```
</details>

<details>
<summary>My environment</summary>

```
Collecting environment information...
PyTorch version: 2.13.0.dev20260516+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: CentOS Stream 9 (x86_64)
GCC version: (GCC) 11.5.0 20240719 (Red Hat 11.5.0-14)
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.34

Python version: 3.13.12 | packaged by Anaconda, Inc. | (main, Feb 24 2026, 16:13:31) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-6.16.1-0_fbk2_hardened_0_gf40efc324cc8-x86_64-with-glibc2.34
Is CUDA available: True
CUDA runtime version: 12.8.93
CUDA_MODULE_LOADING set to: 
GPU models and configuration: GPU 0: NVIDIA H100
Nvidia driver version: 580.126.09
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

CPU:
Architecture:                            x86_64
CPU op-mode(s):                          32-bit, 64-bit
Address sizes:                           52 bits physical, 57 bits virtual
Byte Order:                              Little Endian
CPU(s):                                  46
On-line CPU(s) list:                     0-45
Vendor ID:                               AuthenticAMD
Model name:                              AMD EPYC 9654 96-Core Processor
CPU family:                              25
Model:                                   17
Thread(s) per core:                      1
Core(s) per socket:                      46
Socket(s):                               1
Stepping:                                1
BogoMIPS:                                4792.79
Flags:                                   fpu vme de pse tsc msr pae mce cx8 apic sep mtrr pge mca cmov pat pse36 clflush mmx fxsr sse sse2 ht syscall nx mmxext fxsr_opt pdpe1gb rdtscp lm rep_good nopl cpuid extd_apicid tsc_known_freq pni pclmulqdq ssse3 fma cx16 pcid sse4_1 sse4_2 x2apic movbe popcnt tsc_deadline_timer aes xsave avx f16c rdrand hypervisor lahf_lm cmp_legacy svm cr8_legacy abm sse4a misalignsse 3dnowprefetch osvw perfctr_core ssbd ibrs ibpb stibp ibrs_enhanced vmmcall fsgsbase tsc_adjust bmi1 avx2 smep bmi2 erms invpcid avx512f avx512dq rdseed adx smap avx512ifma clflushopt clwb avx512cd sha_ni avx512bw avx512vl xsaveopt xsavec xgetbv1 xsaves avx512_bf16 clzero xsaveerptr wbnoinvd arat npt lbrv nrip_save tsc_scale vmcb_clean flushbyasid pausefilter pfthreshold v_vmsave_vmload vgif vnmi avx512vbmi umip pku ospke avx512_vbmi2 gfni vaes vpclmulqdq avx512_vnni avx512_bitalg avx512_vpopcntdq la57 rdpid fsrm flush_l1d arch_capabilities
Virtualization:                          AMD-V
Hypervisor vendor:                       KVM
Virtualization type:                     full
L1d cache:                               2.9 MiB (46 instances)
L1i cache:                               2.9 MiB (46 instances)
L2 cache:                                23 MiB (46 instances)
L3 cache:                                736 MiB (46 instances)
NUMA node(s):                            1
NUMA node0 CPU(s):                       0-45
Vulnerability Gather data sampling:      Not affected
Vulnerability Ghostwrite:                Not affected
Vulnerability Indirect target selection: Not affected
Vulnerability Itlb multihit:             Not affected
Vulnerability L1tf:                      Not affected
Vulnerability Mds:                       Not affected
Vulnerability Meltdown:                  Not affected
Vulnerability Mmio stale data:           Not affected
Vulnerability Old microcode:             Not affected
Vulnerability Reg file data sampling:    Not affected
Vulnerability Retbleed:                  Not affected
Vulnerability Spec rstack overflow:      Vulnerable
Vulnerability Spec store bypass:         Vulnerable
Vulnerability Spectre v1:                Vulnerable: __user pointer sanitization and usercopy barriers only; no swapgs barriers
Vulnerability Spectre v2:                Vulnerable; IBPB: disabled; STIBP: disabled; PBRSB-eIBRS: Not affected; BHI: Not affected
Vulnerability Srbds:                     Not affected
Vulnerability Tsa:                       Vulnerable
Vulnerability Tsx async abort:           Not affected

Versions of relevant libraries:
[pip3] mypy_extensions==1.1.0
[pip3] numpy==2.4.4
[pip3] nvidia-cublas==13.1.1.3
[pip3] nvidia-cublas-cu12==12.8.4.1
[pip3] nvidia-cuda-cupti==13.0.85
[pip3] nvidia-cuda-cupti-cu12==12.8.90
[pip3] nvidia-cuda-nvrtc==13.0.88
[pip3] nvidia-cuda-nvrtc-cu12==12.8.93
[pip3] nvidia-cuda-runtime==13.0.96
[pip3] nvidia-cuda-runtime-cu12==12.8.90
[pip3] nvidia-cudnn-cu12==9.10.2.21
[pip3] nvidia-cudnn-cu13==9.20.0.48
[pip3] nvidia-cufft==12.0.0.61
[pip3] nvidia-cufft-cu12==11.3.3.83
[pip3] nvidia-curand==10.4.0.35
[pip3] nvidia-curand-cu12==10.3.9.90
[pip3] nvidia-cusolver==12.0.4.66
[pip3] nvidia-cusolver-cu12==11.7.3.90
[pip3] nvidia-cusparse==12.6.3.3
[pip3] nvidia-cusparse-cu12==12.5.8.93
[pip3] nvidia-cusparselt-cu12==0.7.1
[pip3] nvidia-cusparselt-cu13==0.8.1
[pip3] nvidia-nccl-cu12==2.27.5
[pip3] nvidia-nccl-cu13==2.29.7
[pip3] nvidia-nvjitlink==13.0.88
[pip3] nvidia-nvjitlink-cu12==12.8.93
[pip3] nvidia-nvtx==13.0.85
[pip3] nvidia-nvtx-cu12==12.8.90
[pip3] onnx==1.21.0
[pip3] onnx-ir==0.2.1
[pip3] onnxscript==0.7.0
[pip3] torch==2.13.0.dev20260516+cu132
[pip3] torchao==0.17.0
[pip3] torchaudio==2.11.0.dev20260517+cu130
[pip3] torchdata==0.11.0
[pip3] torchinfo==1.8.0
[pip3] torchmetrics==1.9.0
[pip3] torchrec==1.7.0.dev20260413+cu130
[pip3] torchtune==0.6.1
[pip3] torchvision==0.28.0.dev20260517+cu130
[pip3] triton==3.7.0+git88b227e2
[conda] Could not collect
```
</details>

---
If the repro agent produced an incorrect result (wrong classification, bad repro script, missed the actual bug, etc.), add the label `repro:agent:h100:incorrect` to the issue.
